WordPress/Classes/Services/CommentService.m

```diff
@@ -1144,14 +1145,14 @@ - (void)updateCommentAndSave:(Comment *)comment withRemoteComment:(RemoteComment
 
     // Update depth and hierarchy
     [self setHierarchyAndDepthOnComment:comment withParentComment:parentComment];
-    [[ContextManager sharedInstance] saveContext:self.managedObjectContext];
 }
 
-- (void)mergeHierarchicalComments:(NSArray *)comments forPage:(NSUInteger)page forPost:(ReaderPost *)post onComplete:(void (^)(BOOL includesNewComments))onComplete
+- (BOOL)mergeHierarchicalComments:(NSArray *)comments forPage:(NSUInteger)page forPost:(ReaderPost *)post
```

Comment:
Interesting to see this method changed to remove the completion callback, while `- (void)createReplyForComment:(Comment *)comment content:(NSString *)content completion:(void (^)(Comment *reply))completion` gained one. 

This is a private method, so I assume the reason it changed was to avoid diving even deeper into callback hell?

Reply:
> I assume the reason it changed was to avoid diving even deeper into callback hell?

Yes, correct. Also, since this is a private method, it's okay to not save the "merged changes"  in this function here. Instead, the saving is done at the call site.